lib/perl/Genome/Model/Tools/Bsmap/MethCalcConversionRate.pm

```diff
@@ -0,0 +1,263 @@
+# % Last Change: Thu Jan 15 01:21:18 PM 2015 CST
+package Genome::Model::Tools::Bsmap::MethCalcConversionRate;
+
+use strict;
+use warnings;
+use Genome;
+
+class Genome::Model::Tools::Bsmap::MethCalcConversionRate {
+    is => 'Command',
+    has => [
+        snvs_file => {
+            is => 'String',
+            is_optional => 1,
+            doc => 'Use snvs.hq file to calculate methylation conversion',
+        },
+        model_id => {
```

Comment:
It's generally true that tools in `Genome::Model::Tools` are supposed to be file-based and not interact with the Genome database.  So taking a model ID as an input is breaking that convention.

If that convention needs to be broken then this can take a model as an input instead of an ID by changing its type from 'String' to 'Genome::Model' and removing the `Genome::Model->get()` call on line 127.